src/app/shared/actions/service/order-history.service.ts

```diff
@@ -0,0 +1,162 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { BehaviorSubject, combineLatest, defer, EMPTY, Observable } from 'rxjs';
+import { concatMap, first, map, pluck, tap } from 'rxjs/operators';
+import { WebCryptoApiSignatureProvider } from '../../collector/signature/web-crypto-api-signature-provider/web-crypto-api-signature-provider.service';
+import { DiaBackendAssetRepository } from '../../dia-backend/asset/dia-backend-asset-repository.service';
+import { BUBBLE_DB_URL } from '../../dia-backend/secret';
+import { NetworkAppOrder } from '../../dia-backend/store/dia-backend-store.service';
+import { DiaBackendTransactionRepository } from '../../dia-backend/transaction/dia-backend-transaction-repository.service';
+import { Proof } from '../../repositories/proof/proof';
+import { ProofRepository } from '../../repositories/proof/proof-repository.service';
+import {
+  BubbleCreateNewThingResponse,
+  GetActionsResponse,
+} from './actions.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderHistoryService {
+  public readonly networkActionOrders$ = new BehaviorSubject<
+    BubbleOrderHistoryRecord[]
+  >([]);
+
+  constructor(
+    private readonly httpClient: HttpClient,
+    private readonly webCryptoApiSignatureProvider: WebCryptoApiSignatureProvider,
+    private readonly proofRepository: ProofRepository,
+    private readonly sanitizer: DomSanitizer,
+    private readonly diaBackendTransactionRepository: DiaBackendTransactionRepository,
+    private readonly diaBackendAssetRepository: DiaBackendAssetRepository
+  ) {}
+
+  refresh$() {
+    return combineLatest([
+      this.proofRepository.all$,
+      this.getOrdersHistory$(),
+    ]).pipe(
+      first(),
+      map(([proofs, orders]) => {
+        orders.map(order => {
+          order.assetThumbnailUrl$ = this.fetchThumbnailUrl$(proofs, order);
+        });
+        return orders;
+      }),
+      tap(orders => this.networkActionOrders$.next(orders))
+    );
+  }
+
+  createOrderHistory$(networkAppOrder: NetworkAppOrder, cid: string) {
+    return defer(() =>
+      this.webCryptoApiSignatureProvider.publicKey$.pipe(
+        concatMap(publicKey =>
+          this.httpClient.post<BubbleCreateNewThingResponse>(
+            `${BUBBLE_DB_URL}/api/1.1/obj/order`,
+            {
+              asset_id_text: cid,
+              gas_fee_number: Number(networkAppOrder.fee),
```

Comment:
Using JavaScript number could lose precision for price and gas fee. Also I'm not sure what is the field type and precision on Bubble DB, but we should be careful about it.

Since backend still stored the full precision value, I think it is acceptable to store these values as regular floating point types as long as we don't use them to do calculation that requires 100% precision. But I would strongly recommend writing down this pitfall in network action doc.

Or maybe we could consider storing these value as String. When needed, Bubble side or App could use the full precision string and process them with JavaScript libraries that support big decimals.

Reply:
@shc261392  Bubble db stores network action order price and gas as type `number`.  Also, I tried manually type in the full precision backend return into bubble db, and it lost the precision as well. So I guess even if we could convert it to number without losing precision on app side, Bubble couldn't store it without losing precision. 

So, we either 
1. Keep it as it is now and not worry about having bubble order history db storing price/gas fee with full precision
2. Update the price/gas fee and other `number` fields to `text` to support storing full precision
I need input from @tammyyang 

<img width="380" alt="Screen Shot 2022-03-22 at 4 47 57 PM" src="https://user-images.githubusercontent.com/35753861/159442909-409c61d4-259e-4b2b-93b1-5c07a44b60d8.png">

Reply:
Anyway which can lead us to "consistent and clear results" everywhere is a good method to me.

Reply:
Seems like we couldn't update the field type after the field has been created, I'd say let's go with 1 and query backend using `order_id` if we need to do calculation with full precision.

Reply:
Updated network action dev doc here: https://github.com/numbersprotocol/numbers-dev-doc/blob/main/docs/network_app/develop.md